### ISSUE TITLE
Add "interval": "validation" to scheduler configuration

### ISSUE THREAD

**Issue by de-gozaru** [PyTorchLightning/pytorch-lightning#11224]
## 🚀 Feature

This request concerns the optimizer Scheduler.

For now, the scheduler can be configured like this:
```python
lr_scheduler_config = {
    "scheduler": lr_scheduler,
    "interval": "epoch",  # epoch, step
    "frequency": 1,
    "monitor": "val_loss",
    "strict": True,
    "name": None,
}
```

It would be great to add the option `"interval": "validation"`.

### Motivation

My use case is the following. I want to `.step()` the scheduler after each evaluation step, using an evaluation metric.

Sometimes **(1)** I'm using a bigger dataset, so in the case I use `val_check_interval=0.1`.
and Other times **(2)** I'm using smaller dataset, so I use `check_val_every_n_epoch=5`.

The current interval options are not working for me. `step` will call the scheduler each step, and `epoch` will do it after each epoch, however, for my case **(1)**, the scheduler will be called once every 10 validations, and in **(2)**, the scheduler will be called before the validation, and I will get the error `pytorch_lightning.utilities.exceptions.MisconfigurationException: ReduceLROnPlateau conditioned on metric my_quantity which is not available. Available metrics are: ['train/loss']. Condition can be set using 'monitor' key in lr scheduler dict`.

I hope this is clear for you.

**Comment by rohitgr7** [comment]
@tchaton do you think adding an additional `'validation'` interval would be a good idea? can't think of any configuration to support it with `'step'|'epoch'`. Although there are 2 cases in the above-shared example but I think there can be a case where multiple validation checks can happen in a single training epoch with validation configured to be triggered after every n epochs. So both step and epoch won't work here.

**Comment by gcheron** [comment]
@de-gozaru following the [doc](https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers) you could set `"interval": "step"` and adjust `frequency` to be equal to your evaluation process frequency.
E.g. for **(1)** set:

```
lr_scheduler_config = {
    "scheduler": lr_scheduler,
    "interval": "step",
    "frequency": 0.1*epoch_length,
    "monitor": "val_loss"
}
```

For **(2)** just set `"frequency": 5` and  `"interval": "epoch"`.

**Comment by oiao** [comment]
I'd second this feature. Setting up LR scheduleres that monitor a validation metric and having a fractional `val_check_interval` is currently not easily possible.
